From b1f6c584d90543d72666d7b274208e202e8ce41d Mon Sep 17 00:00:00 2001
From: Rob Herring <ra7055@freescale.com>
Date: Tue, 22 Jan 2008 15:30:00 -0600
Subject: [PATCH] ENGR00060363 make i2c bus ids static

Make the mxc_i2c bus ids static

This fix is necessory to support the i2c new style driver model.
---
 drivers/i2c/busses/mxc_i2c.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/busses/mxc_i2c.c b/drivers/i2c/busses/mxc_i2c.c
index 677b252..5d94f6c 100644
--- a/drivers/i2c/busses/mxc_i2c.c
+++ b/drivers/i2c/busses/mxc_i2c.c
@@ -679,12 +679,12 @@ static int mxci2c_probe(struct platform_device *pdev)
 	 * Set the adapter information
 	 */
 	strcpy(mxc_i2c->adap.name, pdev->name);
-	mxc_i2c->adap.id = id;
+	mxc_i2c->adap.id = mxc_i2c->adap.nr = id;
 	mxc_i2c->adap.algo = &mxc_i2c_algorithm;
 	mxc_i2c->adap.timeout = 1;
 	platform_set_drvdata(pdev, mxc_i2c);
 	i2c_set_adapdata(&mxc_i2c->adap, mxc_i2c);
-	if ((ret = i2c_add_adapter(&mxc_i2c->adap)) < 0) {
+	if ((ret = i2c_add_numbered_adapter(&mxc_i2c->adap)) < 0) {
 		goto err2;
 	}
 
-- 
1.5.4.4

